packages/walletconnect-signing-manager/src/lib/walletconnect/signer.spec.ts

```diff
@@ -0,0 +1,199 @@
+import type { SignerPayloadJSON, SignerPayloadRaw } from '@polkadot/types/types';
+import SignClient from '@walletconnect/sign-client';
+import type { SessionTypes } from '@walletconnect/types';
+
+import { WalletConnectSigner } from './signer';
+
+// Mock the module before importing the class
+jest.mock('@walletconnect/sign-client', () => {
+  const mockRequest = jest.fn();
+  return jest.fn().mockImplementation(() => ({
+    request: mockRequest,
+  }));
+});
+
+describe('WalletConnectSigner', () => {
+  let mockClient: jest.Mocked<SignClient>;
+  let mockSession: jest.Mocked<SessionTypes.Struct>;
+  let walletConnectSigner: WalletConnectSigner;
+
+  beforeEach(() => {
+    mockClient = new SignClient() as jest.Mocked<SignClient>;
+    mockSession = {
+      topic: '59da30fdf6fe82db677a85155d10bbd9f6f5eec94d8e472568e14fd89a33c615',
+      relay: {
+        protocol: 'irn',
+      },
+      expiry: 1716579042,
+      namespaces: {
+        polkadot: {
+          accounts: [
+            'polkadot:6fbd74e5e1d0a61d52ccfe9d4adaed16:5Ci1xTypyBSv1kN5kS5zzFbm1fcfyY6RCqNsDieeYhFjTtVj',
+          ],
+          methods: [
+            'polkadot_signTransaction',
+            'polkadot_signMessage',
+            'personal_sign',
+            'eth_sign',
+            'eth_sendTransaction',
+          ],
+          events: ['chainChanged', 'accountsChanged'],
+          chains: ['polkadot:6fbd74e5e1d0a61d52ccfe9d4adaed16'],
+        },
+      },
+      acknowledged: true,
+      pairingTopic: 'b9eab57c59c35c906a64c71b4a227c33cdf98cc2957e37331552d273ddc867d9',
+      requiredNamespaces: {
+        polkadot: {
+          chains: ['polkadot:6fbd74e5e1d0a61d52ccfe9d4adaed16'],
+          methods: ['polkadot_signTransaction', 'polkadot_signMessage'],
+          events: ['chainChanged', 'accountsChanged'],
+        },
+      },
+      optionalNamespaces: {},
+      controller: 'ba91496933f592bf37d4eb796c084354b90d1bae7eea066d4be59da83f4d9600',
+      self: {
+        publicKey: '64f6d283a62bfad0231ca7fc6dded2d6c491850292bff2fbd47080cd50cc5f3b',
+        metadata: {
+          name: 'Polymesh Signing Manager Test',
+          description: 'Signing Manager Test',
+          url: 'https://example.com',
+          icons: [],
+        },
+      },
+      peer: {
+        publicKey: 'ba91496933f592bf37d4eb796c084354b90d1bae7eea066d4be59da83f4d9600',
+        metadata: {
+          name: 'SubWallet',
+          description: 'React Wallet for WalletConnect',
+          url: 'https://www.subwallet.app/',
+          icons: [
+            'https://raw.githubusercontent.com/Koniverse/SubWallet-Extension/master/packages/extension-koni/public/images/icon-128.png',
+          ],
+        },
+      },
+    } as jest.Mocked<SessionTypes.Struct>;
+
+    walletConnectSigner = new WalletConnectSigner(mockClient, mockSession);
+  });
+
+  test('constructor initializes with provided client and session', () => {
```

Comment:
```suggestion
  it('should initialize with provided client and session', () => {
```
I know they are aliases, but we always use `it` for tests